Bottom: be363292394095bbd8b408bcaea3b34e8acf845f
Top:    a1e17f175f5792855a56ec839ba2b3b0f871f935
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-11-04 02:53:45 +0000

Apparently it might be better to do glVertexPointer last

From: http://sdickinson.com/wordpress/?p=122

""
Make sure you put glVertexPointer at the end as there is a lot of work that
goes on behind the scenes with it, and if it's set at the start, it has to
do all that work for each gl*Pointer call, rather than once at the end.
""


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 38bb75e..ba2659b 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -146,14 +146,14 @@ hidgl_flush_triangles (triangle_buffer *buffer)
     buffer->triangle_array = NULL;
   }
 
-  glVertexPointer (3, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
-                     BUF_OFFSET (0) : buffer->triangle_array + 0);
-
   glTexCoordPointer (2, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
                        BUF_OFFSET (3) : buffer->triangle_array + 3);
 
-  glEnableClientState (GL_VERTEX_ARRAY);
+  glVertexPointer (3, GL_FLOAT, 5 * sizeof (GLfloat), buffer->use_vbo ?
+                     BUF_OFFSET (0) : buffer->triangle_array + 0);
+
   glEnableClientState (GL_TEXTURE_COORD_ARRAY);
+  glEnableClientState (GL_VERTEX_ARRAY);
   glDrawArrays (GL_TRIANGLE_STRIP, 0, buffer->vertex_count);
   glDisableClientState (GL_VERTEX_ARRAY);
   glDisableClientState (GL_TEXTURE_COORD_ARRAY);
